echopype/echodata/echodata.py

```diff
@@ -518,11 +518,21 @@ def _clip_by_time_dim(external_ds, ext_time_dim_name):
         ]
         for platform_var in scalar_vars:
             ext_var = mappings_expanded[platform_var]["external_var"]
-            # Replace the scalar value and add a history attribute
-            platform[platform_var].data = float(extra_platform_data[ext_var].data)
-            platform[platform_var] = platform[platform_var].assign_attrs(
-                **{"history": f"{history_attr}. From external {ext_var} variable."}
-            )
+            if platform_var.startswith("lon") or platform_var.startswith("lat"):
```

Comment:
By reading through what `extra_platform_data` is like, it seems that `platform_var` will always be using the full words "longitude" and "latitude", since they are supposed to be replacing existing data variables already in the `Platform` group, based on the docstring:
```
`extra_platform_data` must be an xarray Dataset. Data is extracted from
`extra_platform_data` by variable name. Only data assigned to a pre-existing
(but possibly all-nan) variable name in Platform will be processed. These
Platform variables include latitude, longitude, pitch, roll, vertical_offset, etc.
See the variables present in the EchoData object's Platform group to obtain a
complete list of possible variables.
```
Let me know if you have a different understanding.

Reply:
Sorry, I think I have misunderstood this part. Updated the code!